src/module_library/fake_solar.h

```diff
@@ -69,8 +69,9 @@ void fake_solar::do_operation() const
     // Calculate the radiation in a way that's good for testing different day lengths, and the switch to dark conditions
 
     // Extract out the DOY and hour
-    int doy = (int)time;
-    double hour = 24.0 * (time - doy);
+    int d =  std::floor(time / 24);
```

Comment:
Since this module now uses `std::floor`, which is part of the `cmath` header library, this file should have `#include <cmath> // for std::floor` up at the top, just before the includes for `module.h` and `state_map.h`.